video/out/vo_kitty.c

```diff
@@ -292,36 +299,42 @@ static bool draw_frame(struct vo *vo, struct vo_frame *frame)
 
 static void flip_page(struct vo *vo)
 {
-    struct priv* p = vo->priv;
-
-    if (p->buffer == NULL)
+    struct priv *p = vo->priv;
+    if (!p->buffer)
         return;
 
-    char *cmd = talloc_asprintf(NULL, TERM_ESC_GOTO_YX, p->top, p->left);
+    bstr cmd = {NULL, 0};
```

Comment:
One last thing you could do is to put `cmd` in `priv`. bstr append will reuse allocated memory, so this would avoid alloc/free on every flip page.

`p->cmd.len = 0;` here. Do the rest the same way. And move `talloc_free(p->cmd.start);` to free_buffers.